xgi/classes/function.py

```diff
@@ -903,3 +905,71 @@ def subfaces(edges, order=None):
 
         faces += faces_to_add
     return faces
+
+
+def complement(H):
+    """Returns the complement of hypergraph H.
+
+    The complement of an hypergraph will have the same nodes (same indexes) and
+    will contain every possible hyperedge linking these nodes, provided it is
+    not already present in the hypergraph and it is not larger than its largest
+    hyperedge.
+    Hyperedges of size one are taken into account.
+
+
+
+    Parameters
+    ----------
+    H : xgi.Hypergraph
+        Hypergraph to complement.
+
+    Returns
+    -------
+    Hc : xgi.Hypergraph
+        Complement of H.
+    """
+
+    from ..generators.classic import empty_hypergraph
+
+    N = len(H.nodes)
+    M = len(H.edges)
+
+    if N == 0:
+        return empty_hypergraph()
+    else:
+        # list the edges in H (as strings such as '001101')
+        I = incidence_matrix(H, sparse=False)
+        edges = set()
+        for row in I.transpose():
+            edge = str()
+            for _01 in row:
+                edge += str(_01)
+            edges.add(edge)
+
+        # list all possible edges of size <= max_edge_size
+        if M == 0:
+            max_edge_size = 0
+        else:  # there is at least one non-empty edge
+            max_edge_size = H.edges.order.max() + 1
```

Comment:
I think this should be equivalent
```suggestion
        max_edge_size = xgi.max_edge_order(H) + 1
```

Reply:
On this : the function `max_edge_order` is defined above in the same python file (`xgi/classes/functions.py`). I could use it directly and write `max_edge_size = max_edge_order(H) + 1` (with the `xgi.` prefix).

Is there a problem with this ?
The function `complement` might not work if it was placed **above** `max_edge_order` in the file, right ?

Reply:
You're right, it's in the same file so you don't need the `xgi.`. The order of the functions in the file should not matter.